### PR TITLE
feat(demo): add an ebike demo

### PR DESCRIPTION
**Pull request by kisvegabor**
This demo was shown on several events by many vendors, and finally could fins some time to clean it up.

https://github.com/user-attachments/assets/28aeddbc-ab02-4620-9c57-87bf4f706d4a

https://github.com/user-attachments/assets/ce707523-7dec-4ad0-8fec-a89d319616b9

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by FASTSHIFT** [comment]
Switch to Chinese language, it should be "简体中文", not "中国人", it should be a translation errors😂.

File: demos/ebike/assets/font_gen.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    os.system(c)
+    print("------------------------------------------------\n")
+
+arabic_font = "../../../scripts/built_in_font/DejaVuSans.ttf"
+arabic_chars_small = "0xFE99,0xFE97,0xFE8D,0xFEBF,0xFE83,0xFE95,0xFE8E,0xFEF3,0xFE8B,0xFEBC,0xFE87,0xFEB3,0xFE98,0xFECB,0xFEDF,0xFED3,0xFEEF,0xFED8,0xFEB1,0xFEAD,0x627,0xFEA9,0xFEF9,0xFED0,0x0644,0xFECF,0xFE94,0x0639,0xFECC,0x0646,0x0648,0x0645,0xFEE3,0xFEE4,0xFEE2,0x0631,0x0633,0x0628,0xFE91,0x0637,0x064A,0x062D,0x0641,0x0636,0x0642,0x0649,0x0638,0x0625,0x0623,0x0621,0xFEE0,0xFEDE,0xFECA,0xFEE8,0xFEE6,0xFEAE,0xFEB4,0xFEB2,0xFE92,0xFE90,0xFEC4,0xFEC2,0xFEF4,0xFEF2,0xFEAA,0xFEDB,0xFEDC,0xFEA0,0xFEA3,0xFEA4,0xFEA2,0xFEF7,0xFEC0,0xFEF0,0xFEE1,0xFEEE,0xFEC9"
+
+arabic_chars_large = "0xFEA9,0xFE8D,0xFEAA,0xFECB,0xFEF9,0xFE95,0xFE8E,0xFEF3,0xFE8B,0xFEBC,0xFEA3,0xFE87"
+
+cjk_font = "../../../scripts/built_in_font/SimSun.woff"
+cjk_chars = "語言標題月日电池今日距离天的速度时间设置蓝牙灯亮度音量最大限度光照强统计三平均高时速。中国人"
```
Comment (FASTSHIFT) [inline demos/ebike/assets/font_gen.py]: "中国人" -> "简体中文", "語言標題"->"语言标题"

File: demos/ebike/translations/zh.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  Battery: 电池
+  Distance today: 今日距离
+  Speed today: 今天的速度
+  Time today: 今天时间
+  SETTINGS: 设置
+  Bluetooth: 蓝牙
+  Lights: 灯
+  Brightness: 亮度
+  Volume: 音量
+  Max. speed: 最大限度。速度
```
Comment (W-Mai) [inline demos/ebike/translations/zh.yml]:
```suggestion
  Max. speed: 最大速度
```

File: demos/ebike/translations/zh.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+zh:
+  Language: 語言
+  Title: 標題
```
Comment (W-Mai) [inline demos/ebike/translations/zh.yml]:
```suggestion
  Language: 语言
  Title: 标题
```

File: demos/ebike/translations/lv_i18n.c [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+static const lv_i18n_lang_t ar_lang = {
+    .locale_name = "ar",
+    .singulars = ar_singulars,
+
+    .locale_plural_fn = ar_plural_fn
+};
+
+static lv_i18n_phrase_t zh_singulars[] = {
+    {"Language", "語言"},
+    {"Title", "標題"},
```
Comment (W-Mai) [inline demos/ebike/translations/lv_i18n.c]:
```suggestion
    {"Language", "语言"},
    {"Title", "标题"},
```

File: demos/ebike/translations/lv_i18n.c [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    {"Battery", "电池"},
+    {"Distance today", "今日距离"},
+    {"Speed today", "今天的速度"},
+    {"Time today", "今天时间"},
+    {"SETTINGS", "设置"},
+    {"Bluetooth", "蓝牙"},
+    {"Lights", "灯"},
+    {"Brightness", "亮度"},
+    {"Volume", "音量"},
+    {"Max. speed", "最大限度。速度"},
```
Comment (W-Mai) [inline demos/ebike/translations/lv_i18n.c]:
```suggestion
    {"Max. speed", "最大速度"},
```

**Review by W-Mai — CHANGES_REQUESTED** [review]
Some translation problems

File: demos/ebike/assets/font_gen.py (hunk 1 above)
Comment (kisvegabor) [reply demos/ebike/assets/font_gen.py]: It's just the list of used characters.

**Comment by kisvegabor** [comment]
@W-Mai Thank you, I've fixed those :slightly_smiling_face:

**Review by FASTSHIFT — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Review by W-Mai — APPROVED** [review]

**Pull request merged**